plugins/source/gcp/client/spec.go

```diff
@@ -9,6 +9,7 @@ type Spec struct {
 	ProjectFilter         string   `json:"project_filter"`
 	BackoffDelay          int      `json:"backoff_delay"`
 	BackoffRetries        int      `json:"backoff_retries"`
+	DiscoveryConcurrency  *int     `json:"discovery_concurrency"`
```

Comment:
I'd suggest to keep it a plain value and check for 0 instead of nil

Reply:
Since 0 would be meaningless (not a good idea to have infinite concurrency?) keeping it plain is an option, OTOH having this as a pointer is a better way of signalling to the user (or code reader/maintainer) that this is an optional value. Can go either way.

Reply:
I was trying to keep the same style as `FolderRecursionDepth` on line 8